openml_OS/models/api/v1/Api_run.php

```diff
@@ -696,35 +717,53 @@ private function run_evaluate() {
     
     $task = $this->Task_inputs->getTaskValuesAssoc($runRecord->task_id);
     if (!array_key_exists('source_data', $task)) {
-      $this->returnError(429, $this->version);
+      // also add error in database
+      $error_code = 429;
+      $data['error'] = $this->load->apiErrors[$error_code];
+      $this->Run_evaluated->replace($data);
+      $this->returnError($error_code, $this->version);
       return;
     }
     if (!array_key_exists('estimation_procedure', $task)) {
-      $this->returnError(430, $this->version);
+      // also add error in database
+      $error_code = 430;
+      $data['error'] = $this->load->apiErrors[$error_code];
+      $this->Run_evaluated->replace($data);
+      $this->returnError($error_code, $this->version);
       return;
     }
     // check if dataset record associated to task exists
     $dataset_record = $this->Dataset->getById($task['source_data']);
     if ($dataset_record === false) {
-      $this->returnError(431, $this->version);
+      // also add error in database
+      $error_code = 431;
+      $data['error'] = $this->load->apiErrors[$error_code];
+      $this->Run_evaluated->replace($data);
+      $this->returnError($error_code, $this->version);
       return;
     }
     // get data quality
     $num_instances_record = $this->Data_quality->getById(array($task['source_data'], 'NumberOfInstances', $this->weka_engine_id));
     if ($num_instances_record === false) {
-      $this->returnError(433, $this->version);
+      // also add error in database
+      $error_code = 433;
+      $data['error'] = $this->load->apiErrors[$error_code];
+      $this->Run_evaluated->replace($data);
+      $this->returnError($error_code, $this->version);
       return;
     }
     // check if estimation procedure record associated to task exists
     $ep_record = $this->Estimation_procedure->getById($task['estimation_procedure']);
     if ($ep_record === false) {
-      $this->returnError(432, $this->version);
+      // also add error in database
+      $error_code = 432;
+      $data['error'] = $this->load->apiErrors[$error_code];
+      $this->Run_evaluated->replace($data);
+      $this->returnError($error_code, $this->version);
       return;
     }
     
     $math_functions = $this->Math_function->getAssociativeArray('name', 'id', 'functionType = "EvaluationFunction"');
-
-    $evaluation_record = $this->Run_evaluated->getById(array($run_id, $eval_engine_id));
     $evaluations_stored = $this->Evaluation->getWhere('source = "' . $run_id . '" AND evaluation_engine_id = "' . $eval_engine_id . '"');
 
     if($evaluation_record && $evaluation_record->error == null) {
```

Comment:
you removed $evaluation_record. This will be undefined and give an error?

Reply:
I moved it up.

Reply:
missed that :)